MetaMask/metamask-mobile
# feat: chain id to hexadecimal format

## **Description**
On this PR we updated the missing dependencies to the metamask controllers v53
* AddressBookController: ^3.0.0
* AssetsController: ^8.0.0
* ControllerUtils: ^4.0.0
* GasFeeController: 6.0.0
* NetworkController: ^9.0.0
* TransactionController: ^6.1.0

(Small minor version of cockatiel update just because it was a missing dependency on solving some conflicts running `yarn add cockatiel` it added with a minor version update, the [changelog](https://github.com/connor4312/cockatiel/commits/master/) seems pretty trivial but we can revert it, let me know your thoughts)

It was needed to change the chain id across the project to use hexadecimal format, and on `Ramp` files was needed to have attention and change the selector to change it back to decimal and see all of the interaction across.

Created a new patch for `SwapsController` to change the chainId to hexadecimal format. This [PR](https://github.com/MetaMask/swaps-controller/pull/184) addresses that change

Created a a core branch to easily reading the assets controller current patch - [v53-mobile-assets-controllers-patch
](https://github.com/MetaMask/core/tree/v53-mobile-assets-controllers-patch)

Implemented migrations for all the state objects that had chain id values and for other breaking changes like `networkDetails` object and `rpcUrl` instead of `rpcTarget`

## **Related issues**

Fixes: #

## **Manual testing steps**
I will mention flows instead of enumerating single steps and upload their recordings, just because it's a wide change around the app

1. Add Tokens with and without auto detection and via dapp
* https://recordit.co/Eun2FJULWt
* https://recordit.co/NZk1ZO5DEV
2. Add NFts with and without auto detection (https://recordit.co/8f2LpXal7p) manually: https://recordit.co/qkrdSOWhJs 
3. Send transaction (https://recordit.co/JAunxorgFI) 
4. Send dapp transaction (https://recordit.co/do2baJ6C4Z)
5. Go to uniswap start a swap flow (https://recordit.co/oZNY4gPivJ)
6. Add and switch between networks (https://recordit.co/2XlUdSWAZB)
7. Add contacts to address book (https://recordit.co/2QNOcKr2tj)
8. Send a swap https://recordit.co/Yi7DpbRwRV
9. Install the previous version, perform a transaction, and install new version the transaction history should be kept (https://recordit.co/zJ1pjUC9ao -> After installing 7.12.0 tag and upgrading to this branch after performing a sepolia transaction, we can see that it persisted on the history view)
10. Show incoming transaction by network - https://recordit.co/MGCP3vugJy
11. Add custom network manually with chain id hexadecimal and decimal format - https://recordit.co/sWAs9XwlxH | https://recordit.co/8ioDINw8vM
12. deeplink to transaction - https://recordit.co/1RiklQ4C80
13. Token detection for optimism and arbitrum and linea - https://recordit.co/AIn88afpSy

## **Screenshots/Recordings**



### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
